scripts/log-tampermonkey-script.js

```
@@ -0,0 +1,29 @@
+const { readFileSync } = require('fs');
+const path = require('path');
+const { getLoaderUrl } = require('../api/get-script');
+
+/**
+ * Generate a Userscript that can be used with tampermonkey to display the IPX widget in a browser. 
+ * The domain is specified with the domain flag when running `npm run gen:tampermonkey`
+ * e.g. --domain=https://google.com
+ * or specify a glob pattern with the pattern flag
+ * e.g. --pattern=https://*.coveo.com/*
+ */
+async function generateTamperMonkeyScript() {
+    const fileName = "tampermonkey.js";
+
+    // Read tampermonkey script template
+    let template = readFileSync(path.resolve(__dirname, '../templates/ipx-tampermonkey.js'), 'utf-8');
+    // Replace loader snipper & glob pattern
+    template = template.replace('{{LOADER_URL}}', getLoaderUrl());
+    
+    // Match all pages in domain
+    const domain = `${process.env.npm_config_domain || '*'}/*`;
```

If no `@match` value is provided, I think we could throw an error with a helpful message:

```
Please specify the site for which the tamper monkey script should be active. e.g. 

npm run log:tampermonkey-script -- google.com
```

Otherwise, if the output defaults to running on all sites, I think some devs might not know that it is possible to restrict the script which could be frustrating.

---

Updated to throw an error with proposed message if no argument is provided.